patterns/disposer.md

```diff
@@ -0,0 +1,84 @@
+# The Promise Disposer Pattern
+
+Often we have to perform resource management in code. For example - release a DB connection:
+
+
+```js
+var users = getDb().then(conn => 
+    conn.query("SELECT name FROM users").then(users => {
```

Comment:
would this be cleaner written using promise caching? i can see a shitstorm break loose about "this is callback hell again", something like

```js
var users = getDb().then(conn => conn.query("SELECT name FROM users"))
users().then(() => conn.release()).catch(() => conn.release())
return users()
```